src/m365/spo/commands/listitem/listitem-retentionlabel-ensure.ts

```diff
@@ -119,73 +120,83 @@ class SpoListItemRetentionLabelEnsureCommand extends SpoCommand {
 
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
     try {
-      const labelInformation = await this.getLabelInformation(args.options, logger);
+      let labelInformation: ListItemRetentionLabel;
+
+      if (args.options.name) {
+        labelInformation = await spo.getWebRetentionLabelInformationByName(args.options.webUrl, args.options.name);
+      }
+      else {
+        labelInformation = await spo.getWebRetentionLabelInformationById(args.options.webUrl, args.options.id!);
+      }
+
+      if (args.options.assetId && !labelInformation.isEventBasedTag) {
```

Comment:
Could you remove these checks? The AssetID field can just be updated, regardless whether the label in question is an event-based label.